### DISCUSSION TITLE
add additional fields to users & tenants table

### DISCUSSION THREAD

**Discussion by denxpert** [JonPSmith/AuthPermissions.AspNetCore#28] [Q&A] [answered]
1) What is the easiest way to add more field to authp.Tenants table? eg. CompanyName, PhoneNumber?

2) Why do we have to keep users in two separate tables authp.AuthUsers and dbo.AspNetUsers when they have to sync pretty much all the time? 2.1) What is the easiest way to add more fields to these tables eg FirstName, LastName?

3) to keep things simple, Is it possible to have all tables to use dbo schema instead of authp. & retail. ?

**Comment by JonPSmith** [comment, marked as answer]
Hi @denxpert,

You can't (and shouldn't) add extra parameters to the AuthP entity classes. Only the AuthP code should have access the AuthP DbContext. But of course you might want to hold extra data for your application, so this is how I do this.

- For a tenant I would create an entity that contains extra data on the company. In Example3 I have a `CompanyTenant` which holds the tenant name, but in a real application I would store more information in a class like that.
- For users I would look at the ASP.NET Core authentication handler to see if it stores that sort of information. Lots of modern authentication handlers do, like Azure AD, [Auth0](https://auth0.com/), Google etc., but others don't. If the authentication handler doesn't help then you need to add your own admin entities to store that data.

> 3.	to keep things simple, Is it possible to have all tables to use dbo schema instead of authp. & retail. ?

The three database parts of an multi-tenant application, that is authentication handler, AuthP's DbContext and your tenant data DbContext, are separated from each other and migrated separately. I set the schema names on the AuthP's DbContext and the tenant data DbContext to make this segregation more obvious and ensure that table names are unique.

You can remove the schema from the tenant data, but the AuthP's DbContext has fixed schema name.
